internal/util/ptrutil/ptr_util.go

```diff
@@ -4,3 +4,23 @@ package ptrutil
 func Ptr[T any](v T) *T {
 	return &v
 }
+
+// ValOrDefault returns a pointer's value if it's non-nil, and the given default
+// value otherwise.
+func ValOrDefault[T any](val *T, defaultVal T) T {
```

Comment:
Similar to `valutil.ValOrDefault`. I find these pretty useful and needed them for an adapter change, so brought 'em in.